subscription-service/src/main/kotlin/com/egm/stellio/subscription/service/SubscriptionService.kt

```diff
@@ -195,19 +202,7 @@ class SubscriptionService(
                     val value = it.value.toSqlValue(it.key)
 
                     val updateStatement = Update.update(columnName, value)
-                    updates.add(
-                        databaseClient.update()
-                            .table("subscription")
-                            .using(updateStatement)
-                            .matching(Criteria.where("id").`is`(subscriptionId))
-                            .fetch()
-                            .rowsUpdated()
-                            .doOnError { e ->
-                                throw BadRequestDataException(
-                                    e.message ?: "Could not update attribute ${it.key}"
-                                )
-                            }
-                    )
+                    updates.add(updateSubscriptionAttribute(subscriptionId, it.key, updateStatement))
```

Comment:
to keep consistency with other similar methods and abstract the r2dbc away from this method, you could just have column name and value as method parameters

Reply:
Yes, it's better

Reply:
OK